src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -783,8 +786,19 @@ private int WritePosixAndGnuSharedFields(Span<byte> buffer)
         // Saves the gnu-specific fields into the specified spans.
         private int WriteGnuFields(Span<byte> buffer)
         {
-            int checksum = WriteAsTimestamp(_aTime, buffer.Slice(FieldLocations.ATime, FieldLengths.ATime));
-            checksum += WriteAsTimestamp(_cTime, buffer.Slice(FieldLocations.CTime, FieldLengths.CTime));
+            int checksum = 0;
+
+            if (_typeFlag is not TarEntryType.LongLink and not TarEntryType.LongPath)
+            {
+                if (_aTime != default)
+                {
+                    checksum += WriteAsTimestamp(_aTime, buffer.Slice(FieldLocations.ATime, FieldLengths.ATime));
```

Comment:
I think it would make sense to move the handling of the `default` value inside the `WriteAsTimestamp` method so it applies to any timestamp written.

Similarly, a `ReadTimestamp` method can be made responsible for handling it, including the mapping of all null bytes (which is now handled in `ReadGnuAttributes`).

Reply:
I thought about this but mtime uses it too. Shouldn't mtime write all zeros if the value to write is MinValue? For that reason, I decided to leave it like this.

Reply:
> Shouldn't mtime write all zeros if the value to write is MinValue?

I don't think that is the case for PAX currently. That is the suggestion.

And also for reading, all zero bytes would be mapped to `default`.

Reply:
I see what you mean. I'd like to double check with other tools to confirm they won't complain if they see mtime set to all nulls.

I opened an issue to track this and test it with other tools: https://github.com/dotnet/runtime/issues/115434